cgen: wrong code generated for option struct with option field

### Describe the bug

cgen is generating wrong code for option struct with option field.

### Expected Behavior

no build error

### Current Behavior

```
/tmp/v_1000/bug.15729669094150551149.tmp.c:12361: error: cannot convert 'struct main__MyStruct' to 'unsigned char'
...
==================
(Use `v -cg` to print the entire error message)

```

### Reproduction Steps

```V

pub struct MyStruct {
pub mut:
	valuea int
	valueb ?int
}
pub struct MyStruct2 {
pub mut:
    valuea int
    valueb ?MyStruct
}

fn main() {
	println(MyStruct2{valuea:1})
}
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

5454562.c9c5fc7

### Environment details (OS name and version, etc.)

OS: linux, Debian GNU/Linux 10 (buster) (VM) (WSL)
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8565U CPU @ 1.80GHz
CC version: cc (Debian 8.3.0-6) 8.3.0

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -323,7 +323,12 @@ fn (mut g Gen) zero_struct_field(field ast.StructField) bool {
 					typ: field.typ
 				}
 				g.write('.${field_name} = ')
-				g.struct_init(default_init)
+				if field.typ.has_flag(.option) {
+					tmp_var := g.new_tmp_var()
+					g.expr_with_tmp_var(default_init, field.typ, field.typ, tmp_var)
+				} else {
+					g.struct_init(default_init)
+				}
 				return true
 			}
 		}

--- a/vlib/v/tests/option_struct_init_test.v
+++ b/vlib/v/tests/option_struct_init_test.v
@@ -1,22 +1,20 @@
-struct Foo {
-	a int
-	b int
-	c int
+pub struct MyStruct {
+pub mut:
+	valueb ?int
 }
 
-struct Holder {
-mut:
-	i int
+pub struct MyStruct2 {
+pub mut:
+	valuea int
+	valueb ?MyStruct
 }
 
-fn add(mut h Holder) ?int {
-	h.i++
-	return h.i
-}
-
-fn test_struct_init_with_multiple_options() {
-	mut h := Holder{}
-	foo := Foo{add(mut h) or { 0 }, add(mut h) or { 0 }, add(mut h) or { 0 }}
-
-	assert foo == Foo{1, 2, 3}
+fn test_main() {
+	a := MyStruct2{
+		valuea: 1
+	}
+	assert a.str() == 'MyStruct2{
+    valuea: 1
+    valueb: Option(error: none)
+}'
 }

